pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -328,6 +328,19 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 	}
 
 	if !infraObj.data.Bool("status", "jobComplete") && infraObj.data.String("status", "failureReason") == "" {
+		job, err := h.getJobFromInfraMachine(infraObj)
+		if apierrors.IsNotFound(err) {
+			// If the job is not found, go ahead and proceed with machine deletion
+			return obj, nil
```

Comment:
I just realized that we probably want to do the following to ensure there are no extraneous objects left behind:
```suggestion
			return obj, h.apply.WithOwner(obj).ApplyObjects()
```